app.py

```diff
@@ -27,24 +38,35 @@
     responses={404: {"description": "Not found"}},
 )
 
-if settings.get("BOOTSTRAP_NODE", False) is True:
-    api_v1.include_router(bootstrap_v1)
-logging.info(f"Bootstrap on this node enabled: {settings.BOOTSTRAP_NODE}")
-logging.info(f"Bootstrap ID: {settings_repository.get_fresh('BOOTSTRAP')}")
 
-is_authentication_enabled: bool = settings.get("AUTH", True)
-if settings.get("TOKENS_NODE", True) is True and is_authentication_enabled:
-    api_v1.include_router(token_v1)
-    logging.info(
-        f"Tokens on this node enabled: {settings.get('TOKENS_NODE', True)}"
-    )
-
-api_v1.include_router(config_v1)
-api_v1.include_router(metadata_v1)
-api_v1.include_router(targets_v1)
-api_v1.include_router(tasks_v1)
-
-rstuf_app.include_router(api_v1)
+def load_endpoints():
+    # load endpoints
+    disabled_endpoints: List[str] = settings.get(
+        "DISABLE_ENDPOINTS", ""
+    ).split(":")
+    if is_authentication_enabled is False:
+        disabled_endpoints.append(f"{api_v1.prefix}{token_v1.prefix}/")
+
+    for v1_endpoint in v1_endpoints:
+        for endpoint_route in v1_endpoint.routes:
+            route = (
+                f"{endpoint_route.methods}{api_v1.prefix}{endpoint_route.path}"
+            )
+            if route in disabled_endpoints:
+                logging.info(f"Disabled endpoint {route}")
+                v1_endpoint.routes.remove(endpoint_route)
+        if f"{api_v1.prefix}{v1_endpoint.prefix}/" in disabled_endpoints:
+            logging.info(
+                f"Disabled endpoint {api_v1.prefix}{v1_endpoint.prefix}/"
+            )
```

Comment:
In your test case, you disable:
`"{'POST'}/api/v1/targets/publish/:/api/v1/targets/"`
I wonder wouldn't the logger log that first you disable `{'POST'}/api/v1/targets/publish/:` and 
then log that you disable the whole `/api/v1/targets/"` endpoint?
At least that seems to me to be the case when I look at the code.

Reply:
Yes, that is correct!
I fixed it. I noticed I was interacting and changing the list. Not a best practice.
Another great spot! 🎉

Reply:
By changing the `endpoint_route` do you mean when you remove it from v1:
`v1_endpoint.routes.remove(endpoint_route)`
or I missed something :D

Reply:
Yes, we shouldn't interact and update the list at the same. It changes the indexes.